mailpoet/tests/integration/Automation/Integrations/WooCommerce/Triggers/AbandonedCartTriggerTest.php

```diff
@@ -0,0 +1,220 @@
+  }
+
+  public function testItUpdatesTaskWhenCartIsUpdated() {
+    $wait = 1;
+    $automation = $this->createAutomation($wait);
+    $this->testee->registerHooks();
+
+    // Add something to the cart.
+    $this->assertIsString(WC()->cart->add_to_cart($this->productId));
+
+    /**
+     * @var ScheduledTaskEntity $scheduled
+     */
+    $scheduled = current($this->tasksRepository->findFutureScheduledByType(AbandonedCartWorker::TASK_TYPE));
+    $this->assertNotNull($scheduled->getCreatedAt());
+
+    //Let some time pass.
+    $createdAt = (new Carbon($scheduled->getCreatedAt()))->subMinute();
+    $this->tasksRepository->persist($scheduled);
+    $this->tasksRepository->flush();
+    $updatedAt = $scheduled->getUpdatedAt();
+    $meta = $scheduled->getMeta();
+    $this->assertIsArray($meta);
+    $productIds = $meta['product_ids'];
+    // Change the cart again.
+    $this->assertIsString(WC()->cart->add_to_cart($this->productId2, 2));
+
+    /**
+     * @var ScheduledTaskEntity[] $newlyScheduled
+     */
+    $newlyScheduled = $this->tasksRepository->findFutureScheduledByType(AbandonedCartWorker::TASK_TYPE);
+    $this->assertCount(1, $newlyScheduled);
+    /**
+     * @var ScheduledTaskEntity $newlyScheduled
+     */
+    $newlyScheduled = current($newlyScheduled);
+    /**
+     * @var \DateTimeInterface $newCreatedAt
+     */
+    $newCreatedAt = $newlyScheduled->getCreatedAt();
+    $this->assertNotSame($createdAt->format('Y-m-d H:i:s'), $newCreatedAt->format('Y-m-d H:i:s'));
+    $this->assertNotSame($updatedAt, $newlyScheduled->getUpdatedAt());
+    $newMeta = $newlyScheduled->getMeta();
+    $this->assertIsArray($newMeta);
+    $this->assertNotSame($productIds, $newMeta['product_ids']);
+    $this->assertSame([$this->productId, $this->productId2], $newMeta['product_ids']);
+  }
+
+  private function createAutomation(int $wait = 1): Automation {
+    $trigger = new Step(
+      'trigger',
+      Step::TYPE_TRIGGER,
+      AbandonedCartTrigger::KEY,
+      [
+        'wait' => $wait,
+      ],
+      [new NextStep('action')]
+    );
+    $action = new Step(
+      'action',
+      Step::TYPE_ACTION,
+      'core:delay',
+      [
+        'delay' => 1,
+        'delay_type' => 'MINUTES',
+      ],
+      []
+    );
+    return (new AutomationFactory())
+      ->withStatusActive()
+      ->addStep($trigger)
+      ->addStep($action)
+      ->create();
+  }
+
+  private function createProduct(string $name, float $price = 1.99): int {
+    $productData = [
+      'post_type' => 'product',
+      'post_status' => 'publish',
+      'post_title' => $name,
+    ];
+    $productId = wp_insert_post($productData);
+    $this->assertIsInt($productId);
+    update_post_meta($productId, '_price', $price);
+    return $productId;
+  }
+
+  public function _after() {
+    wp_set_current_user(0);
+    $this->tasksRepository->truncate();
+    parent::_after(); // TODO: Change the autogenerated stub
```

Comment:
I guess this TODO is not necessary anymore

Reply:
Fixed in https://github.com/mailpoet/mailpoet/pull/5149/commits/e89b970adf97d903bbccb96738105eff5074cf17. Assigning this to QA